### PR TITLE
DetailsList: Add compact mode

### PR DESCRIPTION
**Pull request by leddie24**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [x] Include a change request file if publishing 
- [x] New feature, bugfix, or enhancement
  - [ ] Includes tests
- [x] Documentation update

#### Description of changes

- Adds compact mode to DetailsList
- Style fixes for DetailsList
- Update Check component sizing
- Adds 2 variant examples for DetailList: Document (with compact Toggle), and Compact

#### Focus areas to test

(optional)

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/DetailsList/DetailsList.Props.ts [hunk 1]
```diff
@@ -161,11 +161,14 @@ export interface IDetailsListProps extends React.Props<DetailsList> {
...
   shouldApplyApplicationRole?: boolean;
 
-  /** 
+  /**
    * The minimum mouse move distance to interpret the action as drag event.
    * @defaultValue 5
    */
   minimumPixelsForDrag?: number;
+
+  /** Boolean value to indicate if the component should render in compact mode. Set to false by default */
```
Comment (cschleiden) [inline packages/office-ui-fabric-react/src/components/DetailsList/DetailsList.Props.ts]: Can we explain what compact mode is? And is a boolean good enough, should this be enum for future changes?

File: packages/office-ui-fabric-react/src/components/DetailsList/DetailsRow.scss [hunk 2]
```diff
@@ -1,41 +1,127 @@
 @import '../../common/common';
 
-$rowHeight: 36px;
-$rowVerticalPadding: 10px;
+$rowHeight: 42px;
+$compactRowHeight: 32px;
```
Comment (cschleiden) [inline packages/office-ui-fabric-react/src/components/DetailsList/DetailsRow.scss]: what about allowing users to specify the rowHeight as a prop?

**Comment by micahgodbolt** [comment]
@leddie24 did you work with @phkuo to add all the new semantic variables. Seems this is a pretty big PR for a compact mode.

**Comment by leddie24** [comment]
@micahgodbolt, just spoke with @mikewheaton.  I shouldn't be touching the semantic variables at all apparently? Let me close this and open a new PR with my changes fixed.

File: packages/office-ui-fabric-react/src/components/DetailsList/DetailsRow.scss (hunk 2 above)
Comment (mikewheaton) [inline packages/office-ui-fabric-react/src/components/DetailsList/DetailsRow.scss]: That's a reasonable feature request but is outside of the scope of this change. This is for adding a 'compact' variant to the design system, not opening the door to customizing to different sizes.

**Pull request closed without merging**